newVersion.py

```diff
@@ -5,20 +5,29 @@
 import subprocess
 import sys
 import os
+import textwrap
+
+# iTwin base version to search for. 3.0.x for now.
+itwin_base_version = "3\\.0\\."
+# iTwin Mobile SDK base version to search for. 0.10.x for now.
+mobile_base_version = "0\\.10\\."
 
 def get_executing_directory():
     return os.path.dirname(os.path.realpath(__file__))
 
 def replace_all(filename, replacements):
-    num_replaced = 0
+    num_found = 0
     for line in fileinput.input(filename, inplace=1):
         newline = line
+        found_on_line = False
         for (search_exp, replace_exp) in replacements:
             newline = re.sub(search_exp, replace_exp, newline)
+            if re.search(search_exp, line):
```

Comment:
Rather than repeating the search, can't we just see if `re.sub` changed the line? 

Something like this maybe:
```python
        for (search_exp, replace_exp) in replacements:
            current = re.sub(search_exp, replace_exp, newline)
            if current != newline:
                found_on_line = True
            newline = current
```

Reply:
The old code detected when lines in the file changed. This results in errors if we do a new release without updating the iTwin version. My change here looks for matches to the input regex and counts those, instead of counting the number of lines that actually changed. In theory the re.sub could be inside the re.search if statement, and I will make that change if you really want me to, but your suggested change will result in a refusal at runtime.

Reply:
I moved the re.sub call into the re.search if statement so that it will be more efficient.